vendor/github.com/openshift/kubernetes-drain/drain.go

```diff
@@ -412,9 +413,13 @@ func deleteOrEvictPods(client kubernetes.Interface, pods []corev1.Pod, options *
 
 func evictPods(client typedpolicyv1beta1.PolicyV1beta1Interface, pods []corev1.Pod, policyGroupVersion string, options *DrainOptions, getPodFn func(namespace, name string) (*corev1.Pod, error)) error {
 	returnCh := make(chan error, 1)
+	stopCh := make(chan struct{})
+	var wg sync.WaitGroup
 
 	for _, pod := range pods {
-		go func(pod corev1.Pod, returnCh chan error) {
+		wg.Add(1)
```

Comment:
I'm not crazy about this fix.  It seems like we could have just added a defer close(returnCh) rather than add this new channel?  What do you think?

Reply:
@michaelgugino waitgroup should be used in either case. If we dont want to create a new channel, instead of defer close(returnCh), we could have defer someFunc() 
```golang
defer func() {
close(returnCh)
wg.Wait()
}()
```
And in the goroutine, 
```golang
                                        select {
					case <-returnCh:
						logf(options.Logger, "Received channel close for pod %q. Returning!!!", pod.Name)
						return
					default:
						logf(options.Logger, "error when evicting pod %q (will retry after 5s): %v", pod.Name, err)
						time.Sleep(5 * time.Second)
					}

```